### PR TITLE
fix race condition between catch and finally in do-try.ts

### PR DESCRIPTION
**Pull request by mrjones2014**
- [ ] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- [x] Documentation updated (readme, docs, comments, etc.)
- [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/utilities/core-utils.ts [hunk 1]
```diff
@@ -92,6 +92,18 @@ const _sleep = (milliseconds: number, debug: boolean = false) => {
     );
 };
 
+/**
+ * Block execution for specified number of milliseconds, synchronously.
+ * @param milliseconds the delay
+ */
+const _sleepSync = (milliseconds: number) => {
+    let now = Date.now(),
```
Comment (brandongregoryscott) [inline src/utilities/core-utils.ts:100]: We usually don't define multiple variables with comma syntax - can you update this to be separate? 

```JS
let now = Date.now();
let start = now;
```

File: src/utilities/do-try.test.ts [hunk 2]
```diff
@@ -54,7 +76,7 @@ describe("do-try.ts", () => {
         it("When no errors occur, then catch handler is never called", async () => {
             // Arrange
             const catchHandler = jest.fn();
-            const workload = jest.fn();
+            const workload = async () => jest.fn()();
```
Comment (brandongregoryscott) [inline src/utilities/do-try.test.ts:79]: I think you should be able to create a mock async function with `jest.fn().mockResolvedValue(something)`: https://jestjs.io/docs/en/mock-function-api#mockfnmockresolvedvaluevalue

Looks a bit nicer than this syntax 😛

File: src/utilities/core-utils.ts [hunk 3]
```diff
@@ -134,6 +146,7 @@ export const CoreUtils = {
     objectToArray: _objectToArray,
     range: _.range,
     sleep: _sleep,
+    sleepSync: _sleepSync,
```
Comment (brandongregoryscott) [inline src/utilities/core-utils.ts:149]: While this is only being used in a single test now, we'll want to add some test coverage for it. Please at least add an issue to backfill tests

**Review by brandongregoryscott — CHANGES_REQUESTED** [review]
@mrjones2014 Left some minor feedback

File: src/utilities/do-try.test.ts (hunk 2 above)
Comment (mrjones2014) [reply src/utilities/do-try.test.ts:79]: that's not accomplishing the same thing. The function passed into `try()` has to be an async function.

File: src/utilities/core-utils.ts (hunk 3 above)
Comment (mrjones2014) [reply src/utilities/core-utils.ts:149]: will do

File: src/utilities/do-try.test.ts (hunk 2 above)
Comment (brandongregoryscott) [reply src/utilities/do-try.test.ts:79]: Doesn't it?

```JS
test('async test', async () => {
  const asyncMock = jest
    .fn()
    .mockResolvedValue('default')
    .mockResolvedValueOnce('first call')
    .mockResolvedValueOnce('second call');

  await asyncMock(); // first call
  await asyncMock(); // second call
  await asyncMock(); // default
  await asyncMock(); // default
});
```

Comment (mrjones2014) [reply src/utilities/do-try.test.ts:79]: @brandongregoryscott ah yeah, I could also just make it return a promise. duh. Not sure what I was thinking.

**Review by brandongregoryscott — APPROVED** [review]

**Pull request merged**